src/Orleans/Providers/IOrleansProvider.cs

```diff
@@ -160,4 +161,23 @@ public static TimeSpan GetTimeSpanProperty(this IProviderConfiguration config, s
             return config.Properties.TryGetValue(key, out s) ? TimeSpan.Parse(s) : settingDefault;
         }
     }
+
+    /// <summary>
+    /// Exception thrown whenever a provider has failed to be initialized.
+    /// </summary>
+    [Serializable]
+    public class ProviderInitializationException : OrleansException
```

Comment:
please put all the new exception in one file, better in the same file where all the other  OrleansExceptions are now.